src/heritedcode_api_functions.py

```diff
@@ -0,0 +1,98 @@
+#
+# Copyright (c) nexB Inc. and others. All rights reserved.
+# heritedcode is a trademark of nexB Inc.
+# SPDX-License-Identifier: Apache-2.0
+# See http://www.apache.org/licenses/LICENSE-2.0 for the license text.
+# See https://github.com/nexB/heritedcode for support or download.
+# See https://aboutcode.org for more information about nexB OSS projects.
+#
+
+import requests
+import json
+
+
+class ApiResponse:
+    """
+    this class represents response data structure for returning response
+    from functions.
+    """
+
+    def __init__(
+        self,
+        checksums,
+        data_url,
+        filetype_url,
+        language_url,
+        length,
+        license_url,
+        status,
+    ):
+        """represents diffrent type of checksums like sha1,sha1_git and their values"""
+        self.checksums = checksums
+        """represents data url for available raw data of any code"""
```

Comment:
For these, you should prefer using comments... same for the other ones below. 

Also is this the URL to get the actual file content? If yes it may be best to reuse and "quote" the actual description provided by SWH or something very close to it.

And adding a sample (real) URL  helps a lot understanding what this is about. 

Here is a suggestion:
```suggestion
        # URL to download the content raw bytes. 
        # For example: https://archive.softwareheritage.org/api/1/content/sha1:0001c9f21990e5c61dd92f6a4a93d306987a1a31/raw/
```